ci-operator/step-registry/cucushift/ota/preupgrade/cucushift-ota-preupgrade-commands.sh

```diff
@@ -124,13 +135,19 @@ if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
 fi
 
 set +e
-if [[ "${ENABLE_OTA_TEST}" == "false" ]]; then
-  exit 0
-elif [[ "${ENABLE_OTA_TEST}" == "true" ]]; then
-  extract_oc
-  run_ota_multi_test
-else
-  extract_oc
-  run_ota_single_case ${ENABLE_OTA_TEST}
-fi
-
+oc_extracted="false"
+for c in "OCP-66839" "OCP-24358"; do
```

Comment:
hmm, do we hit any issue with original way by run_ota_single_case and run_ota_multi_test way?
case list in the for loop will be longer and longer.  Especially in run_ota_single_case scenario, the case num should be only 1 in ${ENABLE_OTA_TEST}, but it still travel over all cases. so i think original run_ota_single_case and run_ota_multi_test may distinguish it better.

Reply:
For `run_ota_single_case` scenario, when users enabled a case via ENABLE_OTA_TEST, `run_ota_single_case` is still going to run, then some dirty log are saved.

For `run_ota_multi_test` case, though we did not enable it yet, but I am thinking there would be some consequence issue between multiple cases are enabled. 

The new implementation would allow user to make cases in correct order.

For the introduced loop, it can work with whatever single cases or multiple cases.

Reply:
> run_ota_single_case is still going to run it, then some dirty log are saved.

That's why there is following judgement to check if any func enabled actually. We could improve the log info here to "no case id func found or no need pre step for the case" sth like this.

```
if ! type pre-ocp-"${1}" &>/dev/null; then
        echo "Test Failed: pre-ocp-${1} due to no case id found!" >> "${report_file}"
```

or more better way is to leave a donothing func for such scenario?
```
function pre-OCP-24358(){
  echo "No checkpoints for the case in pre-upgrade step"
}
```

> For the introduced loop, it can work with whatever single cases or multiple cases.
It means we need maintain an extra case list in the code. And it will add unnecessary loop when the list become longer in single case more.

Reply:
Even we keep the original logic, users enabled a case in `ENABLE_OTA_TEST`, the user still need to write a function for the case testing in this step.

For the introduced loop, besides users need to write a function for it, users indeed also need to do one more step,  add the case id in the loop, this extra step is its disadvantage, but on the other side, this will our multiple cases in the future. 

If you prefer the original logic, I am going to revert the change here.

Reply:
> the user still need to write a function for the case testing in this step.

Ah, user need one more step(but not a required step from my side) ,and reader/watcher is more clear to distinguish `no pre/post checkpoints for the case` or `miss checkpoint for some reason` here.

> this will our multiple cases in the future.

Per original discussion, run_ota_multi_test will not need pass trough case id in `ENABLE_OTA_TEST`. It only need run case func in serial and share common vars in run_ota_multi_test.

> I am going to revert the change here.

thanks, i'm not sure if original way will bring any issue in the future, we could improve it at that time.